libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -600,22 +600,30 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 {
 	solUnimplementedAssert(
-		_functionCall.annotation().kind == FunctionCallKind::FunctionCall ||
-		_functionCall.annotation().kind == FunctionCallKind::TypeConversion,
+		_functionCall.annotation().kind != FunctionCallKind::Unset,
 		"This type of function call is not yet implemented"
 	);
 
-	Type const& funcType = type(_functionCall.expression());
+	Type const* funcType;
+
+	if (_functionCall.annotation().kind == FunctionCallKind::StructConstructorCall)
+	{
+		auto const& type = dynamic_cast<TypeType const&>(*_functionCall.expression().annotation().type);
+		auto const& structType = dynamic_cast<StructType const&>(*type.actualType());
+		funcType = structType.constructorType();
+	}
+	else
+		funcType = _functionCall.expression().annotation().type;
 
 	if (_functionCall.annotation().kind == FunctionCallKind::TypeConversion)
```

Comment:
Can you combine this with the previous `if`? Then you can already use the `FunctionType const* functionType` here (of course not assigning in case of type conversion) and do not need the local variable `funcType`.